### ISSUE TITLE
using $ref under a path fails for parameters

### ISSUE THREAD

**Issue by jmelching** [swagger-api/swagger-editor#1972]
- OS: macOS
 - Browser: chrome
 - Version: 72.0.3626.121 
 - Method of installation: latest docker and https://editor.swagger.io/?url=http://localhost:8080/test.yaml
 - Swagger-Editor version: latest
 - Swagger/OpenAPI version: OpenAPI 3.0

### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.0
info:
  title: Test
  description: Test YAML
  version: '1.0'
  contact:
    name: API Support
    email: test@test.com
    url: http://test.com/support
tags:
  - name: test
    description: test
servers:
  - url: http://localhost
paths:
  /employees/{id}:
    $ref: "paths/employees.yaml#/getEmployeeById"
components:
  schemas:
    employee:
      type: object
      properties:
        name:
          type: string
      required:
        - name
  parameters:
    id:
      name: id
      description: test
      in: path
      required: true
      schema:
        type: string
```

paths/employees.yaml
```yaml
getEmployeeById:
  get:
    operationId: getEmployeeById
    description: test
    tags:
     - test
    parameters:
      - $ref: '../test.yaml#/components/parameters/id'
    responses:
      200:
        description: Success
        content:
          application/json:
            schema:
              $ref: '../test.yaml#/components/schemas/employee'
```

### Describe the bug you're encountering
When using $ref under paths: above the editor gives 
```
Semantic error at paths./employees/{id}
Declared path parameter "id" needs to be defined as a path parameter at either the path or operation level
Jump to line 16
```
If I remove the $ref and include the contents of employees.yaml directly, the validation passes.  I also used wework/speccy to validate the yaml and it passes when using $ref.

```bash
± |master {5} S:2 U:3 ?:1 ✗| → docker run -v $PWD/openapi:/project wework/speccy lint test.yaml
Specification is valid, with 0 lint errors
```
### To reproduce...

1. Create test.yaml as above. 
2. Create paths/employees.yaml as above.
3. Navigate to  https://editor.swagger.io/?url=http://localhost:8080/test.yaml
4. Validation error will appear at top.

### Expected behavior
Should be valid according to the openapi 3.0 spec

**Comment by peteclay** [comment]
I have the exact same issue.
The parameter still renders in swagger-ui and passes validation but throws the semantic error.

![image](https://user-images.githubusercontent.com/4447351/55396379-1ef79a00-5587-11e9-8db8-3a35b3a6f74c.png)

Defining the in path parameter in line works fine.

![image](https://user-images.githubusercontent.com/4447351/55396499-58c8a080-5587-11e9-98b0-b7ffffa69c83.png)

**Comment by mariotoffia** [comment]
I've got the (about the) same issue #1976 - I just didn't see this before sumitting my issue.

**Comment by mariotoffia** [comment]
Also same goes for reference in ```info``` of which gives:
```
Errors
Hide
 
Structural error at info
should NOT have additional properties
additionalProperty: $ref
Jump to line 3
Structural error at info
should have required property 'title'
missingProperty: title
Jump to line 3
Structural error at info
should have required property 'version'
missingProperty: version
Jump to line 3
```
the main file:
```yaml
openapi: 3.0.1

info:
  $ref: 'device-local-api/info.yaml'
externalDocs:
```
and the _info.yaml_
```yaml
version: 1.0.0.SNAPSHOT
title: Test title
description: Test
termsOfService: 'http://foobar.io'
contact:
  email: apiteam@foobar.io
license:
  name: Apache 2.0
  url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
```

Works well using my visual studio code plugin but fails with errors above using swagger editor.

Cheers,
 Mario

**Comment by shockey** [comment]
@mariotoffia that error is valid - as the spec indicates, [OpenAPI Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#openapi-object) `info` values can only be [Info Objects](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#infoObject). `$ref`s are only allowed where [Reference Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#referenceObject) is mentioned.

**Comment by mariotoffia** [comment]
@shockey Thanks for that info! :)

**Comment by peteclay** [comment]
Yes, but $refs for URL parameters should work, correct?
(noting that they do "work" in the swagger-ui but throw that error I
screen-shot in the swagger-editor)

**Comment by shockey** [comment]
@peteclay yeah, the case you shared a screenshot of is valid - it's Swagger Editor wrongly not considering the content of the remote reference for validation here!

**Comment by jy95** [comment]
I encountered the same $ref issue with my [specification](https://editor.swagger.io/?url=https://raw.githubusercontent.com/SourceCodeOER/sourcecode_api/refactor_open_api_doc/api.yml)

Other tools don't have this issue (for example [redoc](https://github.com/Redocly/redoc) and [openapi-generator](https://github.com/OpenAPITools/openapi-generator) ) : 

![image](https://user-images.githubusercontent.com/9306961/71105204-6a0d1800-21bd-11ea-8124-e630037ecb7d.png)

**Comment by orinoco** [comment]
Still an issue. It's just an (incorrect) validation error. Beyond clogging up the preview with a bunch of errors that don't actually exist, it's just an inconvenience. Does make it hard to spot real errors though.

Just for reference, the preview still works. no errors in other openapi editors, and it passes validation.

```
npx @openapitools/openapi-generator-cli validate -i openapi.yaml
No validation issues detected.
```

**Comment by MichakrawSB** [comment]
We’ve released Swagger Editor v5! 🎉 Check out the details here: [Inside the New Swagger Editor](https://swagger.io/blog/inside-the-new-swagger-editor/).
We’re closing old issues related to previous versions. If you think any of them are still relevant, please open a new issue – this helps us prioritize what matters most. Thanks for your activity! 🚀
